pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/AvroSchema.java

```diff
@@ -127,7 +127,12 @@ public static void addLogicalTypeConversions(ReflectData reflectData, boolean js
             }
         }
         reflectData.addLogicalTypeConversion(new Conversions.UUIDConversion());
-        reflectData.addLogicalTypeConversion(new Conversions.DecimalConversion());
+        reflectData.addLogicalTypeConversion(new Conversions.DecimalConversion() {
+            @Override
+            public org.apache.avro.Schema getRecommendedSchema() {
+                return LogicalTypes.decimal(100,51).addToSchema(org.apache.avro.Schema.create(org.apache.avro.Schema.Type.BYTES));
```

Comment:
this is hardcoding those parameters.
I am not sure this is good, thinking about future evolutions

Reply:
Yes, this a static function, so we could use some system props here. 
BTW, this is better to support BigDecimal with this default hardcoded settings rather than not support it at all.

Reply:
is there any possible unpleasant side effect for users ?
for instance if I produce a messages using an Avro Schema that has a configuration for this DecimalConversion and then I receive them with a different configuration, can I fall into some dataloss/data corruption case ?

Reply:
we already have "isJsr310ConversionEnabled" and "getAlwaysAllowNull"
why don't we store precision and scale the same way ? this way they will be stored in the registry and consumers will use the right version of the schema